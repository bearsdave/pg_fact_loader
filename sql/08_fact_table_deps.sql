SET client_min_messages TO warning;
--This is for testing functionality of timezone-specific timestamps
SET TIMEZONE TO 'America/Chicago';

--Enable the dep tables - which thereby should be placed first in line!
UPDATE fact_loader.fact_tables ft SET enabled = TRUE
WHERE EXISTS (SELECT 1 FROM fact_loader.fact_table_deps d WHERE d.child_id = ft.fact_table_id);

SELECT fact_loader.worker();
SELECT order_id, customer_id, phone, age, max_order_date, min_total
FROM test_fact.customersorders_fact
ORDER BY order_id;

SELECT fact_loader.worker();
SELECT order_id, customer_id, order_date, total, is_reorder, num_emails, num_read
FROM test_fact.order_emails_fact
ORDER BY order_id;

/****
Nested fact table deps
 */
SELECT fact_loader.worker();
SELECT customer_id, rows_in_customersorders_fact
FROM test_fact.customersorders_summary_fact
ORDER BY customer_id;
