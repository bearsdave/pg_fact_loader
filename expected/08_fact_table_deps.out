SET client_min_messages TO warning;
--This is for testing functionality of timezone-specific timestamps
SET TIMEZONE TO 'America/Chicago';
--Enable the dep tables - which thereby should be placed first in line!
UPDATE fact_loader.fact_tables ft SET enabled = TRUE
WHERE EXISTS (SELECT 1 FROM fact_loader.fact_table_deps d WHERE d.child_id = ft.fact_table_id);
SELECT fact_loader.worker();
 worker 
--------
 t
(1 row)

SELECT order_id, customer_id, phone, age, max_order_date, min_total
FROM test_fact.customersorders_fact
ORDER BY order_id;
 order_id | customer_id |   phone    | age | max_order_date | min_total 
----------+-------------+------------+-----+----------------+-----------
        1 |           1 | 0001234561 |  35 | 04-10-2018     |    100.00
        2 |           3 | 0001234563 |  35 | 04-11-2018     |    200.00
        3 |           5 | 0001234565 |  35 | 04-12-2018     |   2000.00
        4 |           1 | 0001234561 |  35 | 04-13-2018     |    100.00
(4 rows)

SELECT fact_loader.worker();
 worker 
--------
 t
(1 row)

SELECT order_id, customer_id, order_date, total, is_reorder, num_emails, num_read
FROM test_fact.order_emails_fact
ORDER BY order_id;
 order_id | customer_id | order_date |  total  | is_reorder | num_emails | num_read 
----------+-------------+------------+---------+------------+------------+----------
        1 |           1 | 04-10-2018 |  100.00 | f          |          0 |         
        2 |           3 | 04-11-2018 |  200.00 | f          |          0 |         
        3 |           5 | 04-12-2018 | 2000.00 | f          |          1 |        1
        4 |           1 | 04-13-2018 |  100.00 | t          |          0 |         
(4 rows)

/****
Nested fact table deps
 */
SELECT fact_loader.worker();
 worker 
--------
 t
(1 row)

SELECT customer_id, rows_in_customersorders_fact
FROM test_fact.customersorders_summary_fact
ORDER BY customer_id;
 customer_id | rows_in_customersorders_fact 
-------------+------------------------------
           1 |                            2
           3 |                            1
           5 |                            1
(3 rows)

